src/client/service.ts

```diff
@@ -63,3 +66,30 @@ function mapRecord(input: ClientsRecord): Client {
   };
 
 }
+
+export async function addUserPrivilege(ctx: Context, client: Client): Promise<void> {
```

Comment:
Cool, this is well on its way, but needs a few changes:

* Should go in a separate a12n.ts file, we will probably need to add privileges in a bunch of different places so it's good to have this central.

Service files should *never* get `Context`. Context should only ever live in controllers. Instead, pass all the relevant arguments:

```typescript
function addUserPrivilege(principal: string, resource: string, privilege:string): Promise<void> {

}
```

Reply:
so media-api has this line on top of every controller checking the access 
`this.checkOrgAccess(ctx);`
this gave me the inspiration 
but I see your suggestions and will implement!